src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs

```diff
@@ -63,6 +63,9 @@ internal abstract partial class AbstractAddImportFeatureService<TSimpleNameSynta
             ISymbolSearchService symbolSearchService, AddImportOptions options,
             ImmutableArray<PackageSource> packageSources, CancellationToken cancellationToken)
         {
+            if (document is SourceGeneratedDocument)
```

Comment:
This is the actual fix. The rest of the change is just to support triggering code actions in Source Generated doc in test.

Reply:
@mavasani Are we supposed to be triggering code fixes inside source generated documents?

Reply:
would be worth a comment at least explaining why we think this is appropriate.  but it may be clear enough that it's ok to just to have the check.

Reply:
> @mavasani Are we supposed to be triggering code fixes inside source generated documents?

Generally, yes. This is a common request on our team: if you change Syntax.xml in the compiler, that produces new public surface area. Being unable to invoke the Public API fixes is really painful!

Reply:
In that case, I would prefer to _not_ make the change at this location. I would prefer to modify the OOP service to be able to correctly located the source generated document and pass through the change as it normally would.

Reply:
I don't have a very strong preference, but I do feel it's waste of computing resource if we allow the request to go through.

Reply:
what's teh scenario where someone would want to compute 'add import' from *within* an SG document?  It doesn't make a whole lot of sense to me given that you wouldn't be able to actually add the using to the document, since it is readonly.

Reply:
@CyrusNajmabadi This is general service and not a specific code fix provider. The invoking feature could be implemented in a fashion that it knows where to place the update in order for it to apply to the source generated output. Note that it would not be especially difficult to make this change since the implementation is already asynchronous and we can substitute one `GetDocument` call for a `GetDocumentAsync` call.

Reply:
I'd be ok that *if* we get to the point that *some* feature actually wants this functionality because they *then* think they can do something reasonable with teh result, that we then relax this check at that point.

Alternatively, we could just change the *actual* add-import code-fix-provider to bail out early since it def knows it has nothing to do inside an SG doc.  @genlu  would you be ok moving the check there?

Reply:
> @CyrusNajmabadi This is general service

FWIW, i do not believe this to be true.  The name AbstractAddImportFEATUREService was picked because this is effectively the backing service behind the actual 'add using' "feature".  

The actual feature-independent service is just AbstractAddImportsService.  They're named this way to distinguish the two domains.

Reply:
> The actual feature-independent service is just AbstractAddImportsService. They're named this way to distinguish the two domains.

If that description is correct then where the check is seems reasonable to me: the feature doesn't operate, but the underlying engine still can.

Reply:
> If that description is correct then where the check is seems reasonable to me

Yea, then I think having the check in `AbstractAddImportFeatureService.cs` is fine. Also, based on the type name, it's not unreasonable to allow SG document in `IRemoteMissingImportDiscoveryService` as Sam suggested. But I'd prefer not to change it until really needed.

Reply:
I'm not understanding why, if it's so trivial to implement this feature properly in the correct location, we would want to disable it in a different location.

Performance in this case is not a concern. The feature runs fast enough to run regularly in normal documents, so there is no reason to expect manual invocation of the operation in the rarer source generated document would be a problem.

Reply:
> I'm not understanding why, if it's so trivial to implement this feature properly in the correct location, we would want to disable it in a different location.

Because this service is supposed to be implemented to support said codefix and refactoring, and neither of them should run on SG documents, I'd say the proper implementation for the service is to check for SG doc and bail. If we truly intend this to be a general service as @sharwell said, then we should rename the type to  `AbstractMissingImportDiscoveryService`